hyperspy/_components/physical_background.py

```diff
@@ -0,0 +1,205 @@
+		    
+        else: #for 1D
+            g=np.ones([len(model._signal.metadata.Sample.elements)] )
+            for i in range (0,len(w)):            
+                g[i] =w[i].data
+		    
+
+    return g
+
+
+
+def Mucoef(model,quanti): # this function calculate the absorption coefficient for all energy. This, correspond to the Mu parameter in the function
+    w=quanti
+    t=(np.linspace(model._signal.axes_manager[-1].offset,model._signal.axes_manager[-1].size*model._signal.axes_manager[-1].scale,model._signal.axes_manager[-1].size/5))
+    
+    Ac=mass_absorption_mixture(elements=model._signal.metadata.Sample.elements ,weight_percent=w, energies=t)    
+    b=(model._signal.axes_manager.signal_axes[-1].axis)-0.035
+    Ac=np.interp(b,t,Ac)
+    
+    return Ac
+
+
+def Windowabsorption(model,detector): # this function interpolate the detector efficiency based on dictionnary (create from personnal data) and the signal length. This correspond to the Window parameter  
+ 
+    a=np.array(detector_efficiency[detector])
+    b=(model._signal.axes_manager.signal_axes[-1].axis)-0.035
+    x =a[:,0]
+    y = a[:,1]
+    Accc=np.interp(b, x, y)
+    return Accc
+        
+class Physical_background(Component):
+
+    """
+    Background component based on kramer's law and absorption coefficients
+    Attributes
+    ----------
+    a : float
+    b : float
+    E0 : int 
+    """
+
+    def __init__(self, E0, detector, quantification, coefficients=1, Window=0, quanti=0):
+        Component.__init__(self,['coefficients','E0','Window','quanti'])
+
+        self.coefficients._number_of_elements = 2
+        self.coefficients.value = np.ones(2)
+        
+        self.E0.value=E0
+        
+        self._whitelist['quanti'] = quantification
+        self._whitelist['detector'] = detector       
+        self.quanti.value=quanti
+
+        self.Window.value=Window
+
+        self.E0.free=False
+        self.coefficients.free=True
+        self.Window.free=False
+        self.quanti.free=False
+        
+        self.isbackground=True
+
+        # Boundaries
+        self.coefficients.bmin=0
+        self.coefficients.bmax=None
+
+    def initialyze(self): # this function is necessary to initialyze the quant map
+
+        E0=self.E0.value
+
+        self.quanti._number_of_elements=len(self.model._signal.metadata.Sample.elements)
+        self.quanti._create_array()
+
+        if len(self.model.axes_manager.shape)==1:
+            self.quanti.value=Wpercent(self.model,E0,self._whitelist['quanti'])
+        else: 
+            self.quanti.map['values'][:] = Wpercent(self.model,E0,self._whitelist['quanti'])
+            self.quanti.map['is_set'][:] = True
+        
+        self.Window._number_of_elements=len(self.model._signal.axes_manager.signal_axes[-1].axis)
+        self.Window._create_array()
+        self.Window.value=Windowabsorption(self.model,self._whitelist['detector'])
+        
+        return {'Quant map has been created'}
```

Comment:
Is there a reason your function is returning a `set` object?

Reply:
Hmm not really it's more a validation step. 
The quant map can take some time to be created if the dataset is substantial. 
But obviously it's a useless term, just to reassure the user.